### PR TITLE
feat(x/ecocredit): buy/sell query server

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #643 

Add QueryServer implementation for marketplace query API.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/testsuite/queries.go [hunk 1]
```diff
@@ -289,3 +289,348 @@ func (s *IntegrationTestSuite) TestCreditTypeQuery() {
...
+		},
+		// TODO: filtered buy orders required #623
+		//{
+		//	"valid request",
+		//	&ecocredit.QueryBuyOrderRequest{
+		//		BuyOrderId: 1,
+		//	},
+		//	false,
+		//	"",
+		//},
```
Comment (ryanchristo) [inline x/ecocredit/server/testsuite/queries.go:506]: Currently we are not storing direct buy orders and we have not implemented indirect buy orders, so creating a buy order with `Buy` will not store a buy order and we have no buy orders to query.

Comment (ryanchristo) [inline x/ecocredit/server/testsuite/queries.go:506]: Currently we are not storing direct buy orders and we have not implemented indirect buy orders (#623), so creating a buy order with Buy will not store a buy order and we have no buy orders to query.

File: x/ecocredit/server/testsuite/suite.go [hunk 2]
```diff
@@ -973,14 +973,60 @@ func (s *IntegrationTestSuite) TestScenario() {
...
+			{
+				BatchDenom:        batchDenom,
+				Quantity:          "1.0",
+				AskPrice:          &askPrice,
+				DisableAutoRetire: true,
+			},
+		},
+	})
+	s.Require().Nil(err)
+	s.Require().Equal(expectedSellOrderIds, createSellOrder.SellOrderIds)
```
Comment (ryanchristo) [inline x/ecocredit/server/testsuite/suite.go:1003]: This is one valid case that creates a sell order allowing us to query a sell order by ID. The integration tests will still need to be completed (#627).

File: x/ecocredit/server/testsuite/queries.go [hunk 3]
```diff
@@ -289,3 +289,348 @@ func (s *IntegrationTestSuite) TestCreditTypeQuery() {
...
+		},
+		{
+			"invalid order id",
+			&ecocredit.QueryBuyOrderRequest{
+				BuyOrderId: 99,
+			},
+			true,
+			"not found",
+		},
+		// TODO: filtered buy orders required #623
```
Comment (ryanchristo) [inline x/ecocredit/server/testsuite/queries.go:498]: Currently we are not storing direct buy orders and we have not implemented indirect buy orders (#623), so creating a buy order with `Buy` does not store a buy order and we have no buy orders to query.

**Comment by likhita-809** [comment]
can you fix these failing tests @ryanchristo ?

**Comment by ryanchristo** [comment]
> can you fix these failing tests @ryanchristo ?

Just needed to run `go mod tidy` after merging master. Thanks! Ready for review.

**Review by aleem1314 — APPROVED** [review]
LGTM, pre-approving.

Can u also add gRPC REST tests?

**Comment by ryanchristo** [comment]
> Can u also add gRPC REST tests?

Tests added but partially commented out because they are dependent on the creation of buy/sell orders and allowed ask denoms via CLI integration tests and we have not yet implemented CLI commands for marketplace functionality. #615

**Review by likhita-809 — APPROVED** [review]
lgtm

**Pull request merged**